code/modules/cargo/exports/large_objects.dm

```diff
@@ -126,18 +127,25 @@
 								/datum/gas/halon
 								)
 
-	worth += gases[/datum/gas/hypernoblium][MOLES]*5
-	worth += gases[/datum/gas/stimulum][MOLES]*100
-	worth += gases[/datum/gas/freon][MOLES]*15
-	worth += gases[/datum/gas/tritium][MOLES]*5
-	worth += gases[/datum/gas/pluoxium][MOLES]*5
-	worth += gases[/datum/gas/bz][MOLES]*4
-	worth += gases[/datum/gas/miasma][MOLES]*2
-	worth += gases[/datum/gas/hydrogen][MOLES]*1
-	worth += gases[/datum/gas/healium][MOLES]*19
-	worth += gases[/datum/gas/proto_nitrate][MOLES]*5
-	worth += gases[/datum/gas/zauker][MOLES]*1050
-	worth += gases[/datum/gas/helium][MOLES]*6
-	worth += gases[/datum/gas/antinoblium][MOLES]*10
-	worth += gases[/datum/gas/halon][MOLES]*9
+	var/list/gas_prices = list(/datum/gas/bz = 4,
+								/datum/gas/stimulum = 100,
+								/datum/gas/hypernoblium = 5,
+								/datum/gas/miasma = 2,
+								/datum/gas/tritium = 5,
+								/datum/gas/pluoxium = 5,
+								/datum/gas/freon = 15,
+								/datum/gas/hydrogen = 1,
+								/datum/gas/healium = 19,
+								/datum/gas/proto_nitrate = 5,
+								/datum/gas/zauker = 1050,
+								/datum/gas/helium = 6,
+								/datum/gas/antinoblium = 10,
+								/datum/gas/halon = 9
+								)
+
+	for(var/gasID in gases_to_check)
+		C.air_contents.assert_gas(gasID)
+		if(canister_mix[gasID][MOLES] > 0)
+			worth += round((gas_prices[gasID]/k_elasticity) * (1 - NUM_E**(-1 * k_elasticity * canister_mix[gasID][MOLES])))
+
 	return worth
```

Comment:
This is a pre-existing bug but, any call to `.assert_gas/es(...)` needs to be followed up with one of these
```suggestion
	C.air_contents.garbage_collect()
	return worth
```